README.md

```diff
@@ -69,47 +92,71 @@ store.close
 
 ### Delete certificate
 
-Deletes a certificate from certificate store and returns output in string format.
+This method delete certificate from open certificate store.
 
-**Notes: The certificate_name should be valid `CN or Common Name` of the certificate :**
+```
+Input   - Certificate thumbprint
+Return  - True/False 
+```
 
 ```
-certificate_name = 'Root Agency'
 Win32::Certstore.open("Root") do |store|
-    store.delete(certificate_name)
-    store.close
+    store.delete(certificate_thumbprint)
 end
 ```
     or
 ```
 store = Win32::Certstore.open("Root")
-store.delete(certificate_name)
+store.delete(certificate_thumbprint)
 store.close
 ```
 
-### Retrieve certificate
+### Search certificate
 
-Retrieve properties of a certificate from certificate store and returns output in hash format.
+This method search certificate from open certificate store.
+
+```
+Input   - Search Token as: Comman name, Friendly name, RDN and other attributes
+Return  - Matching certificate list 
+```
 
-**Notes: The certificate_name should be valid `CN or Common Name` of the certificate**
+```
+Win32::Certstore.open("Root") do |store|
+    store.search(search_token)
+end
+```
+    or
+```
+store = Win32::Certstore.open("Root")
+store.search(search_token)
+store.close
+```
+
+### Validate certificate
+
+This method validate certificate from open certificate store.
```

Comment:
Validates a certificate in a certificate store is ....... (in what way do we check validity?)